lib/localisation/src/preprocessing/accelerometer.rs

```diff
@@ -1 +1,283 @@
+            return None;
+        }
+
+        Some(clean_accelerometer_data)
+    }
+
+    pub fn check_reliable(&mut self) -> SensorChecks {
+        self.num_outliers_per_accelerometer
+            .iter()
+            .enumerate()
+            .for_each(|(i, val)| {
+                if self.reliable_accelerometers[i]
+                    && val >= &(K_NUM_ALLOWED_ACCELEROMETER_OUTLIERS as i32)
+                {
+                    self.reliable_accelerometers[i] = false;
+                    self.num_reliable_accelerometers -= 1;
+                }
+            });
+
+        if self.num_reliable_accelerometers < K_NUM_ACCELEROMETERS as i32 - 1 {
+            return SensorChecks::Unacceptable;
+        }
+
+        SensorChecks::Acceptable
+    }
+
+    pub fn get_quartiles<const SIZE: usize>(&self, data: &AccelerometerData<SIZE>) -> Quartiles {
+        let mut sorted_data = data.clone();
+        sorted_data.sort_by(|a, b| a.partial_cmp(b).unwrap());
+
+        let quartile_keys: Vec<f32, 3> = Vec::from_slice(&[0.25, 0.5, 0.75]).unwrap();
+        let quartiles: Vec<f32, 3> = quartile_keys
+            .iter()
+            .map(|quartile| {
+                let index_quartile: f32 =
+                    (1.0 + self.num_reliable_accelerometers as f32) * quartile;
+                let index_quartile_floor = index_quartile.floor() as usize - 1;
+                let index_quartile_ceil = index_quartile.ceil() as usize - 1;
+
+                (data.get(index_quartile_floor).unwrap_or(&0.0)
+                    + data.get(index_quartile_ceil).unwrap_or(&0.0))
+                    / 2.0
+            })
+            .collect();
+
+        Quartiles::new(
+            quartiles[0],
+            quartiles[1],
+            quartiles[2],
+            self.num_reliable_accelerometers < K_NUM_ACCELEROMETERS as i32,
+        )
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+
+    #[test]
+    pub fn test_returns_okay() {
+        let mut preprocessor = AccelerometerPreprocessor::new();
+
+        assert_eq!(preprocessor.num_reliable_accelerometers, 4);
+        assert_eq!(
+            preprocessor.reliable_accelerometers,
+            [true; K_NUM_ACCELEROMETERS]
+        );
+        assert_eq!(
+            preprocessor.num_outliers_per_accelerometer,
+            [0; K_NUM_ACCELEROMETERS]
+        );
+
+        let raw_data: RawAccelerometerData<K_NUM_ACCELEROMETERS, K_NUM_AXIS> =
+            RawAccelerometerData::from_slice(&[
+                Vec::from_slice(&[1.0, 2.0, 3.0]).unwrap(), // sqrt(14) = 3.74
+                Vec::from_slice(&[4.0, 5.0, 6.0]).unwrap(), // sqrt(77) = 8.77
+                Vec::from_slice(&[7.0, 8.0, 9.0]).unwrap(), // sqrt(194) = 13.93
+                Vec::from_slice(&[10.0, 11.0, 12.0]).unwrap(), // sqrt(365) = 19.1
+            ])
+            .unwrap();
+
+        let processed_data = preprocessor.process_data(raw_data);
+        assert!(processed_data.is_some());
```

Comment:
It would be good to have a test that checks more than `process_data` just returning `Some(_)`

In this test `processed_data == Some(Vec::from_slice(&[3.7416575, 8.774964, 13.928389, 19.104973]).unwrap());`, if those are the right values then you should assert them, otherwise work out what they should be